level-ws.js

```diff
@@ -59,31 +59,23 @@ function WriteStream (options, db) {
 
   var self = this
 
-  this.on('finish', function f () {
-    if (self._buffer && self._buffer.length) {
-      return self._flush(f)
-    }
+  this.on('finish', function () {
     self.writable = false
     self.emit('close')
   })
 }
 
 inherits(WriteStream, Writable)
 
-WriteStream.prototype._write = function write (d, enc, next) {
+function write (d, enc, next) {
   var self = this
-  if (self._destroyed) return
-
-  if (!self._db.isOpen()) {
-    return self._db.once('ready', function () {
-      write.call(self, d, enc, next)
-    })
-  }
+  if (self.destroyed) return
 
   if (self._options.maxBufferLength &&
       self._buffer.length > self._options.maxBufferLength) {
     self.once('_flush', next)
   } else {
+    // TODO this doesn't seem to make any difference, keeping for now
     if (self._buffer.length === 0) {
       process.nextTick(function () { self._flush() })
     }
```

Comment:
The tests pass if I remove this if-statement, so not _really_ sure if it's needed or not. We probably need more tests for this.

Reply:
By "flush" I mean `db.batch`.